packages/acceptance-tests/pkg-tests-specs/sources/plugins/plugin-npm-cli.test.js

```diff
@@ -0,0 +1,125 @@
+        await writeFile(`${path}/.yarnrc`, `npmAuthToken "${AUTH_TOKEN}"\n`);
+
+        let code;
+        let stdout;
+        let stderr;
+
+        try {
+          ({code, stdout, stderr} = await run(`npm`, `whoami`));
+        } catch (error) {
+          ({code, stdout, stderr} = error);
+        }
+
+        expect({code, stdout, stderr}).toMatchSnapshot();
+      })
+    );
+
+    test(
+      `should print the npm registry username when config has a valid npmAuthIdent`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        await writeFile(`${path}/.yarnrc`, `npmAuthIdent "${AUTH_IDENT}"\n`);
+
+        let code;
+        let stdout;
+        let stderr;
+
+        try {
+          ({code, stdout, stderr} = await run(`npm`, `whoami`));
+        } catch (error) {
+          ({code, stdout, stderr} = error);
+        }
+
+        expect({code, stdout, stderr}).toMatchSnapshot();
+      })
+    );
+
+    test(
+      `should print the npm registry username for a given scope`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        const url = startPackageServer();
+        const rcFileContent = `
+npmScopes:
+  testScope:
+    npmRegistryServer: "${url}"
+
+npmRegistries:
+  "${url}":
+    npmAuthToken: ${AUTH_TOKEN}
+        `;
+
+        await writeFile(`${path}/.yarnrc`, rcFileContent);
+
+        let code;
+        let stdout;
+        let stderr;
+
+        try {
+          ({code, stdout, stderr} = await run(`npm`, `whoami`, `--scope`, `testScope`));
+        } catch (error) {
+          ({code, stdout, stderr} = error);
+        }
+
+        expect({code, stdout, stderr}).toMatchSnapshot();
+      })
+    );
+
+    test(
+      `should throw an error when no auth config is found`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        await expect(run(`npm`, `whoami`)).rejects.toThrowError(/No authentication configured/);
+      })
+    );
+
+    test(
+      `should throw an error when config has an invalid npmAuthToken`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        await writeFile(`${path}/.yarnrc`, `npmAuthToken "${INVALID_AUTH_TOKEN}"\n`);
+        await expect(run(`npm`, `whoami`)).rejects.toThrowError(/Authentication failed/);
+      })
+    );
+
+    test(
+      `should throw an error when config has an invalid npmAuthIdent`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        await writeFile(`${path}/.yarnrc`, `npmAuthIdent "${INVALID_AUTH_IDENT}"\n`);
+        await expect(run(`npm`, `whoami`)).rejects.toThrowError(/Authentication failed/);
+      })
+    );
+
+    test(
+      `should thow an error when invalid auth config is found for a scope`,
+      makeTemporaryEnv({}, async ({ path, run, source }) => {
+        const url = startPackageServer();
+        const rcFileContent = `
+npmScopes:
+  testScope:
+    npmRegistryServer: "${url}"
+
+npmRegistries:
+  "${url}":
+    npmAuthToken: ${INVALID_AUTH_TOKEN}
+        `;
```

Comment:
Not a problem, but fwiw when I need multiple indented lines I tend to use arrays:

```
const rcFileContent = [
  `npmScopes:\n`,
  `  testScope:\n`,
  `    npmRegistryServer: "${url}"\n`,
  ...
].join(``);
```